codex/marketplace.md

```diff
@@ -0,0 +1,157 @@
+which is defined by a Codex node as `expire`,
+the request SHOULD be canceled.
+If canceled, `collateral` SHOULD be returned to any storage nodes and 
+`reward` returned to the requester.
+The requester MAY create a new request with different values to attract storage nodes.
+
+The requester SHOULD initiate the contract with the following values:
+
+```nim
+
+proc requestStorage (): {} =
+  # content identifier
+  byte32 cid = 1
+  # Tokens from the requester to reward storage nodes
+  UInt256 reward = 2
+  # Amount of tokens required for collateral by storage nodes
+  UInt256 collateral = 3
+  # Frequency that proofs are checked by validator nodes
+  byte proofProbability = 4
+  # Amount of desired time for stoageRequest
+  UInt256 duration = 5
+  # The requester can choose amount of spreading of data to storage nodes
+  uint dispersal = 6
+  # Minimum Number of storage hosts 
+  uint nodes = 7
+
+}
+
+```
+
+`cid` 
+
+- MUST be a sha2-256 hash (length of 32 bytes),
+is a content identifier for the requested data to be stored
+- MUST be generated by the requester, a regular node
+
+`reward`
+
+- SHOULD be a token known to the network.
+- it MUST be paid directly to storage nodes that agree to storage contract
+
+`collateral`
+
+All storage nodes MUST provide token collateral before being able to fulfill a storage contract.
+The collateral will be taken to punish a storage node when it does not follow the contract.
+The following conditions MUST be fulfilled in a Codex storage contract
+- failing to provide proof of storage periodically, the RECOMMENDED method for proofs is [Proof-of-Data-Possession](https://hackmd.io/2uRBltuIT7yX0CyczJevYg?view).
+- a portion of `collateral` MUST be offered as a reward to validator nodes,
+and a portion SHOULD be offered as a reward to storage nodes that repair [slots](#slots).
+
+`proofProbability`
+
+Storage nodes are REQUIRED to provide proofs of storage that are verified by validator nodes.
+The requester SHOULD provide the value for the frequency of proofs provided by storage nodes.
+
+`duration`
+
+- Once the `reward` has depleted from periodic storage node payment,
+the storage request SHOULD end.
+The requester MAY renew the storage request by creating a new request with the same `cid` value.
+- Data MAY be considered lost during contract `duration` if storage nodes fail to provide storage proofs and
+no other storage nodes decide to fill empty slots, see [slots](#slots) below.
+
+`dispersal`
+
+- Storage nodes MUST not allowed to fill all slots in a request.
+- Requester SHOULD provide dispersal value
+
+### Slots
+
+After a storage request is made, 
+the requester's Codex node will create slots using erasure coding.
+Each slot will contain a data block being stored.
+- Storage Nodes MUST provide token collateral and proof of storage to fill a slot
+- If storage nodes fail to provide proof of storage in a given period,
```

Comment:
```suggestion
- If storage nodes fail to provide enough proofs of storage,
```
The number of allowed missed proofs is defined at the network-level, meaning it is defined as a parameter to the constructor of the Marketplace contract.